src/ui/components/filters/filterboxcomponent.js

```diff
@@ -200,8 +207,11 @@ export default class FilterBoxComponent extends Component {
           showReset: this.config.resetFilter,
           resetLabel: this.config.resetFilterLabel,
           showExpand: this.config.expand,
-          onChange: filterNode => {
-            this.onFilterNodeChange(i, filterNode);
+          originComponent: this.config.originComponent,
+          onChange: (filterNode, saveFilterNodes, blockSearchOnChange) => {
```

Comment:
One thing that's bugging me about this lambda function a bit is the "saveFilterNodes" param above. It's a little bit misleading. Based on the name, if I set this param to false, I'd expect no filter nodes to be saved/updated. But, that's not necessarily the case, depending on the value of config.searchOnChange.

Reply:
updated to alwaysSaveFilterNodes!